### PR TITLE
Added Random Function to Random Personality 

### LINKED ISSUE #200: Add new hydrate message personality type: Random
## Feature Reasoning

With the introduction of personality types for hydration break messages, it would be nice to have a `random` personality type that essentially chooses a random personality to use for every message for a somewhat "full" or "chaotic" experience.

## Feature Description

The purpose of this feature request is to add a new personality type `random` to the currently existing personality types available for the hydrate break messages. The new personality type should be made selectable via the personality type dropdown menu in the hydrate reminder configuration menu.

It should choose a personality type at random where each personality type has an equal chance of being selected. Afterwards the existing logic to select a specific message for that personality type should be executed.

## Additional Context

As this is not a basic personality type and needs some special logic, it would be good to spend a bit of time working out how to structure the logic appropriately to keep the codebase clean and maintainable. New methods may need to be created to existing classes in order to implement this new feature.

To complete this work, the following two files need to be edited (there may be more as well):
- [HydrateReminderPersonalityType.java](https://github.com/jmakhack/hydrate-reminder/blob/master/src/main/java/com/hydratereminder/HydrateReminderPersonalityType.java)
- [HydrateBreakMessageDictionary.java](https://github.com/jmakhack/hydrate-reminder/blob/master/src/main/java/com/hydratereminder/dictionary/HydrateBreakMessageDictionary.java)

Please also add and/or update unit tests where necessary.

Please refer to [CONTRIBUTING.md](https://github.com/jmakhack/hydrate-reminder/blob/master/CONTRIBUTING.md) for info on how to setup, build, run, and test this project. Feel free to ping @jmakhack in the comments below with any questions if needed. Also, join the [Discord](https://discord.gg/RTgxfFW9mS) to meet and engage with other contributors!

[](https://discord.gg/RTgxfFW9mS)

### PR DESCRIPTION
**Pull request by akshatjainei**
## Associated Issue

Closes #200 

## Implemented Solution
I have included the personality trait in Personality Type file and the solution I have implemented is that when RANDOM is called , a random number is generated from range 1 to 6 and all those number point towards a particular case and generate a random message from it/

### CODE REVIEW

File: src/main/java/com/hydratereminder/dictionary/HydrateBreakMessageDictionary.java [hunk 1]
```diff
@@ -259,6 +260,31 @@ public static String getRandomHydrateBreakMessageForPersonality(HydrateReminderP
...
+                        break;
+                    case 4 :
+                        breakMessage = getRandomBreakMessage(HYDRATE_BREAK_POLITE_TEXT_LIST);
+                        break;
+                    case 5 :
+                        breakMessage = getRandomBreakMessage(HYDRATE_BREAK_MOTIVATIONAL_TEXT_LIST);
+                        break;
+                    case 6 :
+                        breakMessage = getRandomBreakMessage(HYDRATE_BREAK_AGGRESSIVE_TEXT_LIST);
+                        break;
```
Comment (jmakhack) [inline src/main/java/com/hydratereminder/dictionary/HydrateBreakMessageDictionary.java:283]: It looks like a few personality types are left out of the random calculation. Can you make sure to include all personality types?

Comment (akshatjainei) [reply src/main/java/com/hydratereminder/dictionary/HydrateBreakMessageDictionary.java:283]: I will make a commit on your approval @jmakhack

Comment (jmakhack) [reply src/main/java/com/hydratereminder/dictionary/HydrateBreakMessageDictionary.java:283]: @akshatjainei, this looks good for now. Go ahead and commit your changes so I can review the diff on Github.

**Comment by jmakhack** [comment]
It looks like the Codacy Static Code Analysis is failing due to one code style issue reported here: https://app.codacy.com/gh/jmakhack/hydrate-reminder/file/76979739349/issues/source?bid=31418282&fileBranchId=31418282#l271

Can you rename `RandomPersonality()` to `getRandomPersonality()` which should fix the issue?

**Comment by akshatjainei** [comment]
@jmakhack I have renamed it to `getRandomPersonality()`

**Review by jmakhack — APPROVED** [review]

**Pull request merged**